lib/event.ts

```diff
@@ -1,8 +1,24 @@
-export function getEventName(
-  name: string | string[] | undefined,
-  eventTitle: string,
-  eventNameTemplate: string | null
-) {
-  if (!name || !(typeof name === "string")) name = ""; // If name is not set or is not of proper type
-  return eventNameTemplate ? eventNameTemplate.replace("{USER}", name) : eventTitle + " with " + name;
+import { TFunction } from "next-i18next";
+
+export type EventNameObjectType = {
+  attendee: string | string[] | undefined;
+  eventType: string;
+  eventName: string | null;
+  host: string;
+  t: TFunction;
+};
+
+export function getEventName(eventNameObj: EventNameObjectType) {
+  if (!eventNameObj.attendee || !(typeof eventNameObj.attendee === "string")) eventNameObj.attendee = ""; // If name is not set or is not of proper type
+  return !eventNameObj.eventType
+    ? eventNameObj.eventType.replace("{USER}", eventNameObj.attendee)
+    : eventNameObj.eventName +
+        " " +
+        eventNameObj.t("between") +
+        " " +
+        eventNameObj.host +
+        " " +
+        eventNameObj.t("and") +
+        " " +
+        eventNameObj.attendee;
```

Comment:
you can do something like this
```suggestion
eventNameObj.t('event_between_users'. { eventName: eventNameObj.eventName, host: eventNameObj.host, attendee: eventNameObj.attendee })
```

and then in the translation file in `public/static/locales/en/common.json` you would just add
```
"event_between_users": "{{eventName}} between {{host}} and {{attendee}}"
```